cmd/test_runner.go

```diff
@@ -40,28 +44,45 @@ func NewTestRunner(t *testing.T, peerName, testName string) *TestRunner {
 
 // NewTestRunnerWithMockRemoteClient constructs a test runner with a mock remote client
 func NewTestRunnerWithMockRemoteClient(t *testing.T, peerName, testName string) *TestRunner {
-	root, err := repotest.NewMockRepo(peerName, testName)
+	root, err := repotest.NewTempRepo(peerName, testName)
 	if err != nil {
 		t.Fatalf("creating temp repo: %s", err)
 	}
 	root.UseMockRemoteClient = true
 	return newTestRunnerFromRoot(&root)
 }
 
-// NewTestRunnerWithMockRegistry constructs a test runner with a mock registry connection
-func NewTestRunnerWithMockRegistry(t *testing.T, peerName, testName string) *TestRunner {
-	root, err := repotest.NewMockRepo(peerName, testName)
+// NewTestRunnerWithTempRegistry constructs a test runner with a mock registry connection
+func NewTestRunnerWithTempRegistry(t *testing.T, peerName, testName string) *TestRunner {
+	root, err := repotest.NewTempRepo(peerName, testName)
 	if err != nil {
 		t.Fatalf("creating temp repo: %s", err)
 	}
-	root.GetConfig().Registry.Location = "mock"
+
+	reg, teardownRegistry, err := regserver.NewTempRegistry("registry", testName+"_registry")
+	if err != nil {
+		t.Fatalf("creating registry: %s", err)
+	}
+
+	// TODO (b5) - wouldn't it be nice if we could pass the client as an instance configuration
+	// option? that'd require re-thinking the way we do NewQriCommand
+	_, server := regserver.NewMockServerRegistry(*reg)
+
+	runner := newTestRunnerFromRoot(&root)
+	runner.Teardown = func() {
```

Comment:
This works now, but could potentially hide a bug, if there already exists a Teardown function in some future codebase. Should assign `prevTeardown := runner.Teardown`, then called `prevTeardown()` inside the new cleanup function.